staging/src/k8s.io/apiserver/pkg/storage/value/metrics.go

```diff
@@ -135,6 +157,16 @@ func RecordDataKeyGeneration(start time.Time, err error) {
 	dataKeyGenerationLatencies.Observe(sinceInSeconds(start))
 }
 
+// RecordWatchCacheIncrease record watch cache capacity increase operation.
+func RecordWatchCacheCapacityIncrease() {
```

Comment:
nit: maybe instead of having two methods, let's make:

RecordWatchCacheCapacityChange(old, new int) {
   ...
}

And handle the logic of which metric to increase here.

Reply:
Done, inject cacher.objType to watchCache as resource type. I'm not sure is it good enough.